scrapy/core/scraper.py

```diff
@@ -230,10 +246,12 @@ def _log_download_errors(self, spider_failure, download_failure, request, spider
 
         if spider_failure is not download_failure:
             return spider_failure
+        return None
 
-    def _itemproc_finished(self, output, item, response, spider):
+    def _itemproc_finished(self, output: Any, item: Any, response: Response, spider: Spider) -> None:
```

Comment:
I don’t think it needs to be done in this pull request, but I’m thinking that it may be good to have a type alias for all `itemadapter`-supported item types, and use it in scenarios like this one instead of `Any`. It probably makes sense to implement that type alias in `itemadapter`, though.

Reply:
Yeah, assuming it's possible to express.